src/types/index.d.ts

```diff
@@ -180,3 +180,31 @@ export type FilterExpressionRange = {
 };
 
 export type FilterExpressionRangeValue = ['-inf' | number, 'inf' | number];
+
+export interface Item extends Record<string, any> {
+  value: string;
+  is_slotted: boolean;
+  labels: Record<string, unknown>;
+  matched_terms: string[];
+  data?: ItemData;
+  section?: string;
+}
+
+export interface ItemData extends Record<string, any> {
+  url?: string;
+  id?: string;
+  image_url?: string;
+  group_ids?: string[];
+  facets?: Facet[];
+}
+
+export interface Product extends Item {
+  section: 'Products';
+}
+
+export interface SearchSuggestion extends Item {
+  data?: {
+    total_num_results?: number
+  } & ItemData;
+  section: 'Search Suggestions';
```

Comment:
Why do we have a section field? 🤔

Reply:
oooh ur right, that's a good catch. I think in the Autocomplete Ui Library they needed to know, per-item, what kind of item it was (i.e. what section it belongs to).

this implementation has `section` as optional so it should be fine? but it might be confusing to customers using the client-js library :/ Do you think we should remove it?

Reply:
ya 100% agreed. I think that removing `section` from the type here in our client-js library then extending this type over in our constructorio-ui-autocomplete repo makes the most sense :)

Reply:
I'll make the necessary changes, thanks for looking through this :)